solidity/contracts/BurntPixArchives.sol

```diff
@@ -115,12 +169,28 @@ contract BurntPixArchives is LSP8CappedSupply {
         }
         contributions[contributor].iterations += iters;
         // TRANSFER ORIGINAL IF WINNER DETECTED
-        if (contributions[contributor].iterations >= winnerIters && isOriginalUnclaimed()) {
-            IRegistry(registry).transfer(address(this), contributor, burntPicId, true, "");
+        if (
+            contributions[contributor].iterations >= winnerIters &&
+            isOriginalUnclaimed()
+        ) {
+            // todo: have we tested this scenario?
+            IRegistry(registry).transfer(
+                address(this),
+                contributor,
+                burntPicId,
+                true,
+                ""
+            );
         }
         // STORE ARCHIVE IF UNLOCKED
-        if (contributions[contributor].iterations >= IArchiveHelpers(archiveHelpers).fibonacciIterations(contributions[contributor].archiveIds.length + 1)) {
-            bytes32 archiveId = bytes32(++archiveCount);
+        // TODO: Document this here, so the user reading this contract knows when an archived is unlocked
+        if (
+            contributions[contributor].iterations >=
+            IArchiveHelpers(archiveHelpers).fibonacciIterations(
+                contributions[contributor].archiveIds.length + 1 // todo: why +1? to prevent 0?
+            )
+        ) {
+            bytes32 archiveId = bytes32(++archiveCount); // todo: I'm confused about this line
```

Comment:
increments by 1 the count of generated archives and then generates an archive id using that value AFTER the increment happened @Ledespana